sqlmesh/core/signal.py

```diff
@@ -0,0 +1,14 @@
+import abc
+import typing as t
+from datetime import datetime
+
+Interval = t.Tuple[datetime, datetime]
+
+
+class Signal(abc.ABC):
+    @abc.abstractmethod
+    def poll_ready_intervals(self) -> bool | list[Interval]:
+        """Poll the signal and return True if it is ready, otherwise return False."""
```

Comment:
i think we should make it clear that the intervals here need to be aligned with the interval unit, so there's no guessing games on what these values should be

Reply:
nit can we rename this to 

get_intervals 

poll i feel indicates that it will happen many times, but with sqlmesh's way of doing things, it's usually one and done

Reply:
Addressed in https://github.com/TobikoData/sqlmesh/pull/2822/commits/e9d9119249d07679bc4c04c913d4563ffdf8aef7